Closes feature/cleanup

nima opened the pull request.

nima committed:
Race-condition fix for `core:global'

Additionally, the function now takes 3 arguments for arithmetic
operations.

nima committed:
Fix bug reported in CR #46

nima committed:
Complete `::remote:tmux_setup.eval' and utilize

nima committed:
Keep the `.eval', `.json', and `.csv' suffix

Also completed the `util' testing scaffolds.

nima committed:
Only ssh if required (not for localhost)

baua commented on module/unit.sh:
```diff
@@ -243,10 +243,10 @@ function testCoverage() {
                         )
                     )
-                    for fn in ${fns[@]}; do
+                    for fn in "${fns[@]}"; do
```
Why do we use the quotes here?

nima replied:
So that each token is also quoted, for example:

```bash
let -a foo=(
  "a a"
  "b b b"
  "c c c c"
)
```

Without the quote, the other end would read 9 tokens, rather than 3.

baua reviewed (commented):
Good to go.

baua committed:
Add more unit tests to `cpf'

nima committed:
Clean up `remote:mon' usage

nima committed:
Fix `hgd:resolve' regexp (`/../') subfunction

This function reads from `~/.ssh/known_hosts'.  It was broken some
time back and due to missing unit tests / test coverage, never was
picked up.

This commit fixes this issue, and adds the missing unit tests.

Update `hgd' tests using new mock entry point

Other general house-keeping work.

nima committed:
Add `core:bashenv' to allow for environment mocking

Using the new `core:bashenv' function, we can now mock all environment
variables deterministically.  This replaces the old methods:

    SIMPLE_VAR=<env> <cmd>

or:

    BASH_ENV=<(cat <<!
        declare -A COMPLEX_VAR=...
    !
    ) <cmd>

Additionally, the latter of the two previous examples does not work
where `<cmd>' is a function.  The new method has none of these issues.

Finally, fixed a recently introduced bug: new exception-handling code
was not being redirected to stderr.

nima committed:
Add some missing exceptions to `net'

nima committed:
Use `env' to determine which `bash' to run with

This should reduce load/run-time if the preferred bash is not
`/bin/bash', but more importantly will allow for environment-exporting
of non-POSIX keys from the environement; namely those defined with
`declare -a' and `declare -A'.

That is, if `/bin/bash' is ancient, then going from a supported
version, to an unsupported version, and then back to a supported
version is both wasteful, and will garanteed to fail.

Work pertaining to this will be added in this branch.

nima committed:
Renamed user variables to distinguish `var' from non-`var'

nima committed:
Add helper script for local testing

nima commented:
Lots of changes, so another RFR for you @baua :)

nima committed:
Update `codecov' settings
